libraries/tests/AWS.Lambda.Powertools.Parameters.Tests/Provider/ParameterProviderTest.cs

```diff
@@ -718,31 +692,32 @@ public async Task GetMultipleAsync_WhenMaxAgeSet_StoresCachedObjectWithMaxAge()
             MaxAge = duration
         };
 
-        var providerProxy = new Mock<IParameterProviderProxy>();
-        providerProxy.Setup(c =>
-            c.GetMultipleAsync(key, It.IsAny<ParameterProviderConfiguration?>())
-        ).ReturnsAsync(value);
+        var providerProxy = Substitute.For<IParameterProviderProxy>();
+        providerProxy.GetMultipleAsync(key, Arg.Any<ParameterProviderConfiguration?>()).Returns(value);
 
-        var cacheManager = new Mock<ICacheManager>();
-        cacheManager.Setup(c =>
-            c.Get(key)
-        ).Returns(null);
+        var cacheManager = Substitute.For<ICacheManager>();
+        cacheManager.Get(key).Returns(null);
+
+        var powertoolsConfigurations = Substitute.For<IPowertoolsConfigurations>();
 
-        var powertoolsConfigurations = new Mock<IPowertoolsConfigurations>();
-        
         var providerHandler =
-            new ParameterProviderBaseHandler(providerProxy.Object.GetAsync, providerProxy.Object.GetMultipleAsync,
-                ParameterProviderCacheMode.All, powertoolsConfigurations.Object);
-        providerHandler.SetCacheManager(cacheManager.Object);
+            new ParameterProviderBaseHandler(providerProxy.GetAsync, providerProxy.GetMultipleAsync,
+                ParameterProviderCacheMode.All, powertoolsConfigurations);
+        providerHandler.SetCacheManager(cacheManager);
 
         // Act
         var result = await providerHandler.GetMultipleAsync<string>(key, config, null, null);
 
         // Assert
-        cacheManager.Verify(v => v.Get(key), Times.Once);
-        providerProxy.Verify(v => v.GetMultipleAsync(key, It.IsAny<ParameterProviderConfiguration?>()), Times.Once);
-        cacheManager.Verify(v => v.Set(key, value, duration), Times.Once);
-        powertoolsConfigurations.Verify(v => v.SetExecutionEnvironment(providerHandler), Times.Once);
+        cacheManager.Received(1).Get(key);
+        await providerProxy.Received(1).GetMultipleAsync(key, Arg.Any<ParameterProviderConfiguration>());
+        cacheManager.Received(1).Set(key, Arg.Is<Dictionary<string, string?>>(x=> 
```

Comment:
Logic is very different than the current implementation. value Dictiionary is passed with Guids